### PR TITLE
Blog on migrating from kube-dns to CoreDNS

### PR DESCRIPTION
**Pull request by rajansandeep**

### CODE REVIEW

File: content/blog/migrate-kubedns-to-coredns-in-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+~~~
+$ minikube addons enable coredns
+coredns was successfully enabled
+~~~
+
+
+## CoreDNS in kube-up
+
+[Kube-up](https://kubernetes.io/docs/getting-started-guides/scratch/) is another way to start a Kubernetes cluster, now mostly used for deploying Kubernetes in GCE for end-to-end (e2e) testing purposes.
```
Comment (fturib) [inline content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: Maybe we should inform here about the other variable to have DNS started. But sure that the reader understand that both of these variables need to he TRUE to have CoreDNS installed.

File: content/blog/migrate-kubedns-to-coredns-in-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+CoreDNS can be installed as the default DNS service, you need to set the environment variable `CLUSTER_DNS_CORE_DNS` to `true`.
+
+## CoreDNS in Kops
+
+Currently, Kops is set to include [CoreDNS as an option](https://github.com/kubernetes/kops/pull/4041) to be installed instead of kube-dns in v1.10.
```
Comment (fturib) [inline content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: v1.10 of kops ?

File: content/blog/migrate-kubedns-to-coredns-in-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+~~~text
+   .:53 {
+        errors
+        health
+        kubernetes cluster.local  in-addr.arpa ip6.arpa {
+           upstream  8.8.8.8 8.8.4.4
+           pods insecure
+           fallthrough in-addr.arpa ip6.arpa
+        }
+       federation cluster.local {
```
Comment (fturib) [inline content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: missing space to align federation with above "{"

File: content/blog/migrate-kubedns-to-coredns-in-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+~~~
+
+To enable CoreDNS, run the following command:
+> NOTE: Ensure to disable kube-dns after enabling CoreDNS.
+
+~~~
+$ minikube addons enable coredns
+coredns was successfully enabled
+~~~
+
```
Comment (fturib) [inline content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: And then what happen ? 
when will CoreDNS really be installed ? - is it possible to switch from kubedns to coredns ?

Comment (fturib) [reply content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: and what if you enable both "kube-dns" and "coredns" ?

File: content/blog/migrate-kubedns-to-coredns-in-kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
++++
+title = "Migration from Kube-dns to CoreDNS"
+description = "A guide to migration from kube-dns to CoreDNS in Kubernetes"
+tags = ["Kubernetes", "Service", "Discovery", "Kube-DNS", "DNS", "Migration", "Documentation"]
+date = "2018-04-29T10:12:43-00:00"
+author = "sandeep"
++++
+
+CoreDNS is currently a Beta feature in Kubernetes and on course to being graduated to [General Availability (GA) for Kubernetes 1.11](https://github.com/kubernetes/community/pull/1956).
+This means that CoreDNS will be available as the standard in Kubernetes via the installation toolkits such as kubeadm, kube-up, minikube and kops.
```
Comment (johnbelamaric) [inline content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: s/as the standard/as a standard/

File: content/blog/migrate-kubedns-to-coredns-in-kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+This document will guide you to migrating the DNS service from CoreDNS to kube-dns when using the various tools available to spin up a Kubernetes cluster.
+
+## Installing CoreDNS via Kubeadm
+
+There is an extensive guide on how to install CoreDNS instead of kube-dns via Kubeadm available [here](https://coredns.io/2018/01/29/deploying-kubernetes-with-coredns-using-kubeadm).
+Currently, from Kubernetes v1.10, CoreDNS supports the translation of the kube-dns configmap to CoreDNS configmap.
```
Comment (johnbelamaric) [inline content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: s/Currently, from/From/

Comment (johnbelamaric) [reply content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: s/configmap/ConfigMap/g

File: content/blog/migrate-kubedns-to-coredns-in-kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+That is, if you had configured `stubdomains`, `upstreamnameservers` and `federation` via the kube-dns configMap, it will now be translated automatically to the equivalent CoreDNS configmap during when choosing to install CoreDNS using `kubeadm upgrade`.
```
Comment (johnbelamaric) [inline content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: more ConfigMap capitalization throughout...

File: content/blog/migrate-kubedns-to-coredns-in-kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+`Stubdomain` and `upstreamnameserver` in kube-dns translates to the [`proxy`](https://coredns.io/plugins/proxy/) in CoreDNS.
```
Comment (johnbelamaric) [inline content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: You should make a note about possibly using forward instead.

File: content/blog/migrate-kubedns-to-coredns-in-kubernetes.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+        cache 30
+        proxy . 2.3.4.5
+    }
+~~~
+
+
+
+## Installing CoreDNS via Minikube.
+
+CoreDNS is available in the `addon manager` disabled by default.
```
Comment (johnbelamaric) [inline content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: s/disabled/and is disabled/

File: content/blog/migrate-kubedns-to-coredns-in-kubernetes.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+- coredns: disabled
+- heapster: disabled
+- efk: disabled
+- ingress: disabled
+- default-storageclass: enabled
+- storage-provisioner: enabled
+~~~
+
+To enable CoreDNS, run the following command:
+> NOTE: Ensure to disable kube-dns after enabling CoreDNS.
```
Comment (johnbelamaric) [inline content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: s/Ensure/Be sure/

File: content/blog/migrate-kubedns-to-coredns-in-kubernetes.md [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+[Kube-up](https://kubernetes.io/docs/getting-started-guides/scratch/) is another way to start a Kubernetes cluster, now mostly used for deploying Kubernetes in GCE for end-to-end (e2e) testing purposes.
+CoreDNS can be installed as the default DNS service, you need to set the environment variable `CLUSTER_DNS_CORE_DNS` to `true`.
+
+## CoreDNS in Kops
+
+Currently, Kops is set to include [CoreDNS as an option](https://github.com/kubernetes/kops/pull/4041) to be installed instead of kube-dns in v1.10.
+
+## Installing CoreDNS via other methods
+
+For users keen to install CoreDNS in place of kube-dns but are not using `kubeadm`, `minikube`, `kube-up`, or `kops`, there is an existing document available [here](https://github.com/coredns/deployment/tree/master/kubernetes), which will help you to migrate from kube-dns to CoreDNS.
```
Comment (johnbelamaric) [inline content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: s/but are/but who are/

Comment (johnbelamaric) [reply content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: s/there is an existing document [here]/there are instructions in the [CoreDNS deployment repository]/

File: content/blog/migrate-kubedns-to-coredns-in-kubernetes.md [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+It is recommended to delete the kube-dns deployment after deploying CoreDNS. Otherwise, if both CoreDNS and kube-dns are running, queries may randomly hit either CoreDNS or kube-dns.
```
Comment (johnbelamaric) [inline content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: s/It is recommended to delete/You should delete/

File: content/blog/migrate-kubedns-to-coredns-in-kubernetes.md (hunk 4 above)
Comment (rajansandeep) [reply content/blog/migrate-kubedns-to-coredns-in-kubernetes.md]: Addressed.

**Comment by rajansandeep** [comment]
@johnbelamaric PTAL

**Comment by miekg** [comment]
Put this live on Monday +  tweet?

**Pull request merged**